File: frontend/src/types/node.ts
export type NodeType = 'AGENT' | 'DATASOURCE' | 'TOOL' | 'RULE'

export interface AppNode {
  id: string
  workspace_id: string
  type: NodeType
  label: string
  x_position: number
  y_position: number
  width: number | null
  height: number | null
  created_at: string
  updated_at: string
}

export interface NodeProperty {
  id: string
  node_id: string
  key: string
  value: string
  value_type: string
  created_at: string
  updated_at: string
}

File: backend/src/models/node.rs
use chrono::{DateTime, Utc};
use serde::Serialize;
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, FromRow, Serialize)]
pub struct Node {
    pub id: Uuid,
    pub workspace_id: Uuid,
    #[sqlx(rename = "type")]
    #[serde(rename = "type")]
    pub node_type: String,
    pub label: String,
    pub x_position: f64,
    pub y_position: f64,
    pub width: Option<f64>,
    pub height: Option<f64>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
